test/ServerTest.cpp

```diff
@@ -316,21 +316,22 @@ TEST(ServerTest, configurePinnedResultWithName) {
 
   // Test with no pinNamed value - should not modify qec
   std::optional<std::string> noPinNamed = std::nullopt;
-  Server::configurePinnedResultWithName(noPinNamed, true, *qec);
+  Server::configurePinnedResultWithName(noPinNamed, std::nullopt, true, *qec);
   EXPECT_FALSE(qec->pinResultWithName().has_value());
 
   // Test with pinNamed and valid access token - should set the pin name
   std::optional<std::string> pinNamed = "test_query_name";
-  Server::configurePinnedResultWithName(pinNamed, true, *qec);
+  Server::configurePinnedResultWithName(pinNamed, std::nullopt, true, *qec);
   EXPECT_TRUE(qec->pinResultWithName().has_value());
-  EXPECT_EQ(qec->pinResultWithName().value(), "test_query_name");
+  EXPECT_EQ(qec->pinResultWithName().value().name_, "test_query_name");
 
   // Reset for next test
   qec->pinResultWithName() = std::nullopt;
 
   // Test with pinNamed but invalid access token - should throw exception
   AD_EXPECT_THROW_WITH_MESSAGE(
-      Server::configurePinnedResultWithName(pinNamed, false, *qec),
+      Server::configurePinnedResultWithName(pinNamed, std::nullopt, false,
+                                            *qec),
```

Comment:
I think we also need a test that includes a not `std::nullopt` value for the new parameter.